pkg/buildkit/worker/pool_test.go

```diff
@@ -390,39 +325,26 @@ func TestPoolGetFailedScaleUp(t *testing.T) {
 	})
 
 	wp := NewPool(fakeClient, testConfig, SyncWaitTime(50*time.Millisecond))
-	defer wp.Close()
-	wp.Start(ctx)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go wp.Start(ctx)
 
 	addr, err := wp.Get(ctx, owner)
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	leaseChannel := make(chan result)
-	go func() {
-		addr, err := wp.Get(ctx, owner)
-		leaseChannel <- result{addr, err}
-	}()
+	expected := "tcp://buildkit-0.buildkit.test-namespace:1234"
+	if addr != expected {
+		t.Errorf("did not received correct lease: %s expected, %s actual", expected, addr)
+	}
```

Comment:
IMO this isn't the same test, though I can understand why this is changed. I think the intention was:

1. Lease
2. Lease
3. Release number 1
4. Validate lease number 2 received the existing address

Reply:
See it now. Will fix it.

Reply:
Although the test is testing whether it failed the scale up, not get/release/get which is different.

Reply:
This test makes less and less sense.
There is no validation that a scale up was even attempted given the sts is set to 0 replicas.
A get/release/get test should be separate and I will add it to TestRelease since its part of testing the expected behavior.

Reply:
Ok, I see what is supposed to happen... Will add more verification of expectations which will make this more clear.